src/connection/host_lookup.c

```diff
@@ -0,0 +1,435 @@
+
+static int host_conf_element_Count = 0;
+static int host_conf_Use = -1;	/* -1 is unknown, 0 is not using user hosts, 1 is using user hosts */
+
+// *INDENT-OFF*
+static std::unordered_map <std::string, std::string> user_host_Map;
+// *INDENT-ON*
+
+static int host_conf_load ();
+static struct hostent *host_lookup_internal (const char *hostname, struct sockaddr *saddr, LOOKUP_TYPE lookup_case);
+
+static struct hostent *
+host_lookup_internal (const char *hostname, struct sockaddr *saddr, LOOKUP_TYPE lookup_case)
+{
+  static struct hostent hp;
+  int i, find_index = -1;
+
+  char addr_transform_buf[IPADDR_LEN];
+  char addr_transform_buf2[IPADDR_LEN];
+  struct sockaddr_in *addr_trans = NULL;
+
+  if (host_conf_element_Count == 0)
+    {
+      host_conf_element_Count = host_conf_load ();
+      if (host_conf_element_Count <= 0)
+	{
+	  return NULL;
+	}
+    }
+
+  addr_trans = (struct sockaddr_in *) saddr;
+
+  assert (((hostname != NULL) && (saddr == NULL)) || ((hostname == NULL) && (saddr != NULL)));
+
+  if (lookup_case == IPADDR_TO_HOSTNAME)
+    {
+      if (inet_ntop (AF_INET, &addr_trans->sin_addr, addr_transform_buf, sizeof (addr_transform_buf) + 1) == NULL)
+	{
+	  return NULL;
+	}
+
+    }
+
+/*Look up in the hostent_List*/
+  for (i = 0; i < host_conf_element_Count; i++)
+    {
+      if ((lookup_case == HOSTNAME_TO_IPADDR) && (strcmp (hostname, hostent_List[i].hostname) == 0))
+	{
+	  find_index = i;
+	  break;
+	}
+
+      if ((lookup_case == IPADDR_TO_HOSTNAME) && (strcmp (addr_transform_buf, hostent_List[i].ipaddr) == 0))
+	{
+	  find_index = i;
+	  break;
+	}
+    }
+
+  if (find_index < 0)
+    {
+      return NULL;
+//this should be set to error message
+    }
+
+  if (inet_pton (AF_INET, hostent_List[find_index].ipaddr, addr_transform_buf2) < 1)
+    {
+      return NULL;
+    }
+
+  /*set the hostent struct for the return value */
+  hp.h_addrtype = AF_INET;
+  hp.h_length = 4;
+  hp.h_name = (char *) malloc (sizeof (char) * HOSTNAME_BUF_SIZE);
+  hp.h_aliases = (char **) malloc (sizeof (char *));
+  hp.h_aliases[0] = (char *) malloc (sizeof (char) * HOSTNAME_BUF_SIZE);
+  hp.h_addr_list = (char **) malloc (sizeof (char *) * HOSTNAME_BUF_SIZE);
+  hp.h_addr_list[0] = (char *) malloc (sizeof (char) * IPADDR_LEN);
+
+  strcpy (hp.h_name, hostent_List[find_index].hostname);
+  strcpy (hp.h_aliases[0], hostent_List[find_index].hostname);
+  memcpy (hp.h_addr_list[0], addr_transform_buf2, sizeof (hp.h_addr_list[0]));
+
+  return &hp;
+}
+
+static int
+host_conf_load ()		//set hash table to discover error
+{
+  FILE *fp;
+  char file_line[HOSTNAME_BUF_SIZE + 1];
+  char line_buf[HOSTNAME_BUF_SIZE + 1];
+  char host_conf_file_full_path[PATH_MAX];
+  char *hosts_conf_dir;
+
+  char *token;
+  char *save_ptr_strtok;
+  /*delimiter */
+  char *delim = " \t\n";
+
+  int host_count = 0;
+
+  /*False, if hostent_List[index].hostname is set */
+  bool storage_flag = IPADDR_STORE;
```

Comment:
Assigning the IPADDR_STORE value of storage_flag does not seem appropriate.
It does not actually store the IP Addr.